src/ligature.rs

```diff
@@ -0,0 +1,146 @@
+use skrifa::{
+    charmap::Charmap,
+    raw::{
+        tables::gsub::{CoverageTable, LigatureMarker, LigatureSet, SubstitutionSubtables},
+        types::Offset16,
+        ArrayOfOffsets, FontRef, ReadError, TableProvider, TableRef,
+    },
+    GlyphId,
+};
+use std::{cell::RefCell, iter};
+
+use crate::error::IconResolutionError;
+
+pub struct Ligature {
+    // Includes all components, not just 1..N.
+    pub components: Vec<GlyphId>,
+    pub glyph: GlyphId,
+}
+
+type LigatureInnerIter<'a> = Vec<(
+    GlyphId, // First gid in ligature components.
+    RefCell<Box<dyn Iterator<Item = Result<TableRef<'a, LigatureMarker>, ReadError>> + 'a>>,
```

Comment:
I don't think you want refcell, box, or dyn. It will be slightly fiddly to wire, let me try to see if I can send you a PR to tweak this.

Reply:
Proposed fix in https://github.com/googlefonts/sleipnir/commit/cc724029fea314bbb35e0704622e92b3d3796625